lib/workers/repository/update/branch/index.ts

```diff
@@ -73,17 +76,37 @@ async function deleteBranchSilently(branchName: string): Promise<void> {
 }
 
 export interface ProcessBranchResult {
+  configAndManagersHash?: string;
   branchExists: boolean;
   prBlockedBy?: PrBlockedBy;
   prNo?: number;
   result: BranchResult;
 }
 
+function canSkipBranchUpdateCheck(
+  branchName: string,
+  branchCache: BranchCache,
+  configAndManagersHash: string
+): boolean {
+  const branchCommit = getBranchCommit(branchName);
+
+  return (
+    branchCommit === branchCache.sha &&
+    configAndManagersHash === branchCache.configAndManagersHash
+  );
+}
 export async function processBranch(
-  branchConfig: BranchConfig
+  branchConfig: BranchConfig,
+  branchCache: BranchCache
```

Comment:
Is there something that prevents `branchCache` to be obtained inside `processBranch()` body rather than passing it as the parameter?

Reply:
That was the initial flow. We discussed it here https://github.com/renovatebot/renovate/pull/16661#discussion_r928218370